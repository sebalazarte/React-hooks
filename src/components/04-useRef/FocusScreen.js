import React, {useRef} from 'react'
import '../02-useEfect/effects.css';



export const FocusScreen = () => {

    const inputRef = useRef();
    
    const handleClick = () => {
        // document.querySelector('input').focus();
        inputRef.current.focus();
    }

    return (
        <div>
            <h1>Focus screen</h1>
            <hr/>
            <input
                ref={inputRef}
                className="form-control"
                placeholder="Su nombre"
            />
            <button 
                className="btn btn-outline-primary mt-2"
                onClick={handleClick}>
                Focus
            </button>
        </div>
    )
}
