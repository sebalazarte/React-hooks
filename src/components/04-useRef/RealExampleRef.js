import React, { useState } from 'react'
import '../02-useEfect/effects.css';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';


export const RealExampleRef = () => {

    const [show, setShow] = useState(false);

    return (
        <div>
            <h1>RealExampleRef</h1>
            <hr />
            {show && <MultipleCustomHooks />}

            <button
                className="btn btn-outline-primary mt-2"
                onClick={() => { setShow(!show) }}>
                Show / Hide
            </button>
        </div>
    )
}
