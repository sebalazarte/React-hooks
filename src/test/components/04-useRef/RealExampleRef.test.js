import { shallow } from "enzyme";
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";


describe('Real Example Ref', () => {
    
    const wrapper = shallow(<RealExampleRef />);

    test('debe mostrar correctamente el componente ', () => {
        
        expect(wrapper).toMatchSnapshot();       
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);

    });

    test('debe mostrar el componente MultipleCustomHooks ', () => {

        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    })

});