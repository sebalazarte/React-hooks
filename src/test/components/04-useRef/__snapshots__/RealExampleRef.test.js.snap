// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Real Example Ref debe mostrar correctamente el componente  1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary mt-2"
    onClick={[Function]}
  >
    Show / Hide
  </button>
</div>
`;
